src/main/Store.java

```diff
@@ -0,0 +1,131 @@
+package main;
+
+/**
+ * General class that keeps actions of the store
+ *
+ * @author Radu Popescu
+ */
+
+import product.AnimalProduct;
+import product.Product;
+import product.VegetalProduct;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.*;
+
+public class Store {
+    private final String[] animalProductNames = new String[]{"milk", "eggs", "cheese", "yogurt", "sour cream"};
+    public List stockItemList = new ArrayList();
+    public List soldItemList = new ArrayList();
+
+    public int readInt(String intMessageRequest) {
+        Scanner sc = new Scanner(System.in);
+        System.out.println(intMessageRequest);
+        int value = sc.nextInt();
+        return value;
+    }
+
+    public String readString(String stringMessageRequest) {
+        Scanner sc = new Scanner(System.in);
+        System.out.println(stringMessageRequest);
+        String text = sc.nextLine();
+        return text;
+    }
+
+    public Product addProduct(String name, int price, String date, int weight) throws ParseException {
```

Comment:
a better name would be "createProduct", you're creating and returning the product, not adding it anywhere